File: repository-local/src/main/java/com/andyanika/translator/repository/local/ModelsAdapter.java
package com.andyanika.translator.repository.local;

import com.andyanika.translator.common.models.FavoriteModel;
import com.andyanika.translator.common.models.LanguageCode;
import com.andyanika.translator.common.models.TranslateDirection;
import com.andyanika.translator.common.models.TranslateResult;
import com.andyanika.translator.repository.local.model.WordFavoriteModel;
import com.andyanika.translator.repository.local.model.WordModel;

import javax.inject.Inject;

class ModelsAdapter {
    @Inject
    ModelsAdapter() {
    }

    TranslateResult toTranslationResult(WordModel dbModel) {
        TranslateDirection<LanguageCode> direction = new TranslateDirection<>(LanguageCode.valueOf(dbModel.languageSrc), LanguageCode.valueOf(dbModel.languageDst));
        return new TranslateResult(dbModel.textSrc, dbModel.textDst, direction);
    }

    FavoriteModel toTranslationRowModel(WordModel dbModel) {
        TranslateResult translateResult = toTranslationResult(dbModel);
        return new FavoriteModel(dbModel.id, translateResult, true);
    }

    FavoriteModel toTranslationRowModel(WordFavoriteModel dbModel) {
        TranslateDirection<LanguageCode> direction = new TranslateDirection<>(LanguageCode.valueOf(dbModel.languageSrc), LanguageCode.valueOf(dbModel.languageDst));
        TranslateResult translateResult = new TranslateResult(dbModel.textSrc, dbModel.textDst, direction);
        return new FavoriteModel(dbModel.id, translateResult, dbModel.favoriteId > 0);
    }

    WordModel toWordModel(TranslateResult translateResult) {
        return new WordModel(translateResult.textSrc, translateResult.textDst, translateResult.direction.src.toString(), translateResult.direction.dst.toString());
    }
}

File: repository-local/src/main/java/com/andyanika/translator/repository/local/ModelsAdapter.kt
package com.andyanika.translator.repository.local

import core.models.FavoriteModel
import core.models.LanguageCode
import core.models.TranslateDirection
import core.models.TranslateResult
import com.andyanika.translator.repository.local.entity.WordFavoriteEntity
import com.andyanika.translator.repository.local.entity.WordEntity

class ModelsAdapter {
    fun toTranslationResult(dbEntity: WordEntity): TranslateResult {
        val direction =
            TranslateDirection(LanguageCode.valueOf(dbEntity.languageSrc), LanguageCode.valueOf(dbEntity.languageDst))
        return TranslateResult(dbEntity.textSrc, dbEntity.textDst, direction)
    }

    fun toTranslationRowModel(dbEntity: WordEntity): FavoriteModel {
        val translateResult = toTranslationResult(dbEntity)
        return FavoriteModel(dbEntity.id, translateResult, true)
    }

    fun toTranslationRowModel(dbEntity: WordFavoriteEntity): FavoriteModel {
        val direction =
            TranslateDirection(LanguageCode.valueOf(dbEntity.languageSrc), LanguageCode.valueOf(dbEntity.languageDst))
        val translateResult = TranslateResult(dbEntity.textSrc, dbEntity.textDst, direction)
        return FavoriteModel(dbEntity.id, translateResult, dbEntity.favoriteId > 0)
    }

    fun toWordModel(translateResult: TranslateResult): WordEntity {
        return WordEntity(
            translateResult.textSrc,
            translateResult.textDst,
            translateResult.direction.src.toString(),
            translateResult.direction.dst.toString()
        )
    }
}
